src/api.js

```diff
@@ -464,56 +511,104 @@ var remoteAuthentication = function(options, entry) {
           return false;
         }
 
-        // If we're not given an array, we assume it's a set of parameters that
-        // must be used to parameterize the original scopesets
-        if (!(scopesets instanceof Array)) {
-          var params = scopesets;
-          scopesets = _.cloneDeepWith(entry.scopes, function(scope) {
+        req.hasAuthed = false;
+        let missing = [];
+        let hasConditional = false;
+
+        var _splatParams = (scope) => scope.replace(/<([^>]+)>/g, function(match, param) {
+          var value = _.at(params, param)[0];
+          if (value !== undefined) {
+            return value;
+          }
+          missing.push(match); // If any are left undefined, we can't be done yet
+          return match;
+        });
+
+        var _expandExpressionTemplate = (template) => {
```

Comment:
Yeah, let's do this in another file... that way we can have specific unit tests for it too...